dallinger/deployment.py

```diff
@@ -384,7 +385,12 @@ def ensure_constraints_file_presence(directory: str):
     constraints_path = Path(directory) / "constraints.txt"
     requirements_path = Path(directory) / "requirements.txt"
     if constraints_path.exists():
-        return
+        is_up_to_date = (
```

Comment:
As we discussed on Slack, we're going to opt to raise an error here and prompt the user to update (regenerate) their constraints.txt file.